src/overlays/actors/ovl_En_Daiku2/z_en_daiku2.c

```diff
@@ -22,56 +41,531 @@ const ActorInit En_Daiku2_InitVars = {
+    this->unk_278 = this->actor.params & 0x7F;
+    this->unk_27A = (this->actor.params >> 7) & 0x3F;
+    this->unk_258 = func_8013D648(globalCtx, this->unk_27A, 0x3F);
+    this->unk_280 = (this->actor.params >> 0xF) & 1;
+    Actor_SetScale(&this->actor, 0.01f);
+    if (this->unk_280 == 0) {
+        if (day == 3) {
+            Actor_MarkForDeath(&this->actor);
+            return;
+        }
+        if (this->unk_278 == 0x7F) {
+            this->unk_278 = -1;
+        } else if (Flags_GetSwitch(globalCtx, this->unk_278)) {
+            this->unk_25C = this->unk_258->count - 1;
+            func_80BE61D0(this);
+            Math_Vec3f_Copy(&this->actor.world.pos, &this->unk_268);
+        }
+    } else if (day != 3) {
+        Actor_MarkForDeath(&this->actor);
+        return;
+    }
+
+    this->actor.gravity = -3.0f;
+    Math_Vec3f_Copy(&this->unk_268, &this->actor.world.pos);
+    func_80BE65B4(this, globalCtx);
+}
+
+void EnDaiku2_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnDaiku2* this = THIS;
+
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+}
+
+void func_80BE6408(EnDaiku2* this, s32 arg1) {
+    f32 sp34 = 1.0f;
+
+    this->unk_276 = arg1;
+    this->unk_284 = SkelAnime_GetFrameCount((AnimationHeaderCommon*)D_80BE792C[this->unk_276]);
+    if (this->unk_276 == 3) {
+        sp34 = 2.0f;
+    }
+    SkelAnime_ChangeAnim(&this->skelAnime, D_80BE792C[this->unk_276], sp34, 0.0f, this->unk_284,
+                         D_80BE7958[this->unk_276], -4.0f);
+}
+
+s32 func_80BE64C0(EnDaiku2* this, GlobalContext* globalCtx) {
+#include "overlays/actors/ovl_En_Bom/z_en_bom.h"
+
+    EnBom* bomb;
+    Vec3f sp30;
+
+    Math_Vec3f_Copy(&sp30, &this->actor.world.pos);
+    Math_Vec3f_Copy(&this->actor.world.pos, &this->actor.home.pos);
+    bomb = (EnBom*)func_800BE0B8(globalCtx, &this->actor, -1, ACTORCAT_EXPLOSIVES, gGameInfo->data[2407] + 240.0f);
+    Math_Vec3f_Copy(&this->actor.world.pos, &sp30);
+    if ((this->unk_278 >= 0) && !Flags_GetSwitch(globalCtx, this->unk_278) && (bomb != NULL) &&
+        (bomb->actor.id == ACTOR_EN_BOM)) {
+        if (bomb->unk_1F9 == 0) {
+            this->actor.textId = 0x32D3;
+        } else {
+            this->actor.textId = 0x32D4;
+        }
+        return 1;
+    }
+    return 0;
+}
+
+void func_80BE65B4(EnDaiku2* this, GlobalContext* globalCtx) {
+    switch (gSaveContext.day - 1) {
+        case 0:
+            this->unk_28A = 0;
+            if (gSaveContext.weekEventReg[64] & 2) {
+                this->unk_28A = 6;
+            }
+            func_80BE6408(this, 8);
+            break;
+
+        case 1:
+            this->unk_28A = 2;
+            if (gSaveContext.weekEventReg[64] & 4) {
+                this->unk_28A = 7;
+            }
+            func_80BE6408(this, 8);
+            break;
+
+        case 2:
+            this->unk_28A = 4;
+            this->unk_264 = 1.0f;
+            func_80BE6408(this, 10);
+            break;
+    }
+
```

Comment:
```suggestion
    switch (gSaveContext.day) {
        case 1:
            this->unk_28A = 0;
            if (gSaveContext.weekEventReg[64] & 2) {
                this->unk_28A = 6;
            }
            func_80BE6408(this, 8);
            break;

        case 2:
            this->unk_28A = 2;
            if (gSaveContext.weekEventReg[64] & 4) {
                this->unk_28A = 7;
            }
            func_80BE6408(this, 8);
            break;

        case 3:
            this->unk_28A = 4;
            this->unk_264 = 1.0f;
            func_80BE6408(this, 10);
            break;
    }

```

Reply:
This doesn't work because it's not a jump table, it's just an if/elseif chain.

Reply:
Did you tried? This usually works, and IDO should optimize it to what you have.